hyperspy/component.py

```diff
@@ -910,20 +907,18 @@ def _component2plot(self, axes_manager, out_of_range2nans=True):
         if axes_manager is not self.model.axes_manager:
             old_axes_manager = self.model.axes_manager
             self.model.axes_manager = axes_manager
-            self.charge()
```

Comment:
I guess it's one of the plotting bugs? Now I can see what might be a problem, I'll add the lines shortly